lib/index.js

```diff
@@ -457,6 +458,15 @@ function nop() {
 function pipeReadable(xs, stream) {
     // write any errors into the stream
     xs.on('error', writeStreamError);
+
+    // stream is the thing being returned, and shouldn't be writable
+    // but pipe needs something writable
+    proxy = function () {
```

Comment:
Having a proxy is a good idea, but you need to define a actual Proxy class (preferably outside of the `pipeReadable` function that forwards calls to `write`/`end`/`on`/`emit` rather than setting `stream` as the prototype. The way it is now will work all the way until a variable needs to be reassigned.

Here's an illustration of the problem.
```javascript
function Stream() {
    this.a = 0;
}

Stream.prototype.add = function () {
    this.a++;
};

var stream = new Stream();

function Proxy() {
}

Proxy.prototype = stream;

var proxy = new Proxy();

console.log('stream', stream.a); // => stream 0
console.log('proxy', proxy.a);   // => proxy 0

proxy.add();
console.log('stream', stream.a); // => stream 0 (!)
console.log('proxy', proxy.a);   // => proxy 1
```